DataCollector/DoctrineDataCollector.php

```diff
@@ -222,15 +222,26 @@ public function getGroupedQueries()
             });
             $groupedQueries[$connection] = $connectionGroupedQueries;
         }
+
         foreach ($groupedQueries as $connection => $queries) {
             foreach ($queries as $i => $query) {
-                $groupedQueries[$connection][$i]['executionPercent'] = $query['executionMS'] / $totalExecutionMS * 100;
+                $groupedQueries[$connection][$i]['executionPercent'] =
+                    $this->executionTimePercentage($query['executionMS'], $totalExecutionMS );
             }
         }
 
         return $groupedQueries;
     }
 
+    private function executionTimePercentage($executionTimeMS, $totalExecutionTimeMS)
+    {
+        if ($totalExecutionTimeMS === 0.0 || $totalExecutionTimeMS === 0) {
```

Comment:
s it safe to compare float values like this? Maybe we can use very small epsilon value e.g. if ($totalExecutionTimeMS < $eps) where $eps = 1e-5 or something like this

Reply:
Yes, this comparison looks dangerous.